client/python/gradio_client/client.py

```diff
@@ -781,7 +809,7 @@ def __init__(self, client: Client, fn_index: int, dependency: dict):
         self.api_name: str | Literal[False] | None = (
             "/" + api_name if isinstance(api_name, str) else api_name
         )
-        self.use_ws = self._use_websocket(self.dependency)
+        self.protocol = self._get_protocol(self.dependency)
```

Comment:
Just merged a PR that causes some conflicts. But all the prototcol related stuff for backwards compatibility with websockets/http can be added to `EndpointV3Compatibility` class